animate/metric.py

```diff
@@ -84,12 +84,12 @@ def __init__(self, function_space, *args, **kwargs):
         if isinstance(fs.dof_count, Iterable):
             raise ValueError("Riemannian metric cannot be built in a mixed space.")
         self._check_space()
-        rank = len(fs.dof_dset.dim)
-        if rank != 2:
-            raise ValueError(
-                "Riemannian metric should be matrix-valued,"
-                f" not rank-{rank} tensor-valued."
-            )
+        # rank = len(fs.dof_dset.dim)
+        # if rank != 2:
+        #     raise ValueError(
+        #         "Riemannian metric should be matrix-valued,"
+        #         f" not rank-{rank} tensor-valued."
+        #     )
```

Comment:
I guess we *could* drop this check but we'd need to make sure we do check the rank is 2 for various methods for which it would presumably break, e.g., metric intersection.

Reply:
Uncommented in e652186f3837642a84c77e4ae34e6789af98eaf6 since the metric advection now solves for the entire metric tensor